packages/grid/test/data-provider.common.js

```diff
@@ -247,6 +255,15 @@ describe('data provider', () => {
         { path: 'name', direction: 'asc' },
       ]);
     });
+
+    it('should not request again when resolving with an empty array', async () => {
```

Comment:
One of my earlier fix attempts passed all other tests but it caused an infinite data request loop so I added this test to make sure some future change will not cause one either.